### ISSUE TITLE
`--verifytypes` doesn't find packages linked with `pth` files

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2670]
**Describe the bug**
After setting up a poetry project and activating the virtual environment, `pyright --verifytypes` gives an error for the project's package:

```
> pyright --verifytypes pyright_verifytypes_poetry                                                                                                               
Package name: "pyright_verifytypes_poetry"                                                                                                                                                                                      
Package directory: ""                                                                                                                                                                                                           
  error: Package "pyright_verifytypes_poetry" cannot be resolved 
```

Other packages in the virtual environment are found, though:

```
> pyright --verifytypes pyparsing
Package name: "pyparsing"
Package directory: "/Users/bryan/Projects/pyright-verifytypes-poetry/.venv/lib/python3.9/site-packages/pyparsing"
  error: No py.typed file found
```

**To Reproduce**
1. Clone https://github.com/bryanforbes/pyright-verifytypes-poetry
2. Run `poetry install`
3. Run `poetry run pyright --verifytypes pyright_verifytypes_poetry`

**Expected behavior**
`pyright` finds the package and produces the report

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.193

**Comment by erictraut** [comment]
Here's what's happening. When pyright attempts to determine the current Python environment and the import resolution paths within it, it executes `python3` with a small script that returns the `sys.path` array. For security reasons, this script explicitly removes the working directory from `sys.path` prior to importing `json` or other libraries it needs. Otherwise this could be an attack vector. 

In this particular case, I'm guessing that you're running `poetry run pyright --verifytypes pyright_verifytypes_poetry` from the `pyright-verifytypes-poetry` directory, so that's the working directory. The "pth" file points back to this same directory, and that directory is therefore included in the `sys.path` array. However, our script is then removing it from the `sys.path` array because it's the working directory. 

If you run `poetry shell` (to activate the venv) and then switch to a different directory and run `pyright --verifytypes pyright_verifytypes_poetry`, it should work as expected because the working directory has changed.

@jakebailey, you added the code that removes the working directory from sys.path for security reasons. Do you have any thoughts on how to fix this? Maybe the script should remove the working directory but then add it back after it imports `json`? At that point, it should no longer be a security issue, right?

**Comment by erictraut** [comment]
@jakebailey, here's a PR for you to review when you have a minute. I've confirmed this fixes the reported bug. https://github.com/microsoft/pyright/pull/2677

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.194, which I just published.
